src/oop/object_interactions/medium/q5/MainTest.java

```diff
@@ -0,0 +1,208 @@
+                {"Sarah", 2, "John", 0},
+                {"Amir", 1, "Sam", 2},
+                {"John", 0, "Jack", 0},
+        };
+        for (int num = 0; num < test.length; num++) {
+            String name = (String) test[num][0];
+            int eyes = (int) test[num][1];
+            Object[][] arguments = {
+                    {name, String.class},
+                    {eyes, int.class}
+            };
+            Object classInstance = testClass.createInstance(arguments);
+            _assertEquals(name, testClass.getFieldValue(classInstance, myNameAttributeName),
+                    "You have modified the provided constructor on the " + className + " class. Please revert it to the original state.");
+            _assertEquals(eyes, testClass.getFieldValue(classInstance, numEyesAttributeName),
+                    "You have modified the provided constructor on the " + className + " class. Please revert it to the original state.");
+            String updateName = (String) test[num][2];
+            int updateEyes = (int) test[num][3];
+            Object output = testClass.callMethod(getMyNameMethodName, classInstance);
+            _assertEquals(name, output,
+                    "You have modified the provided " + getMyNameMethodName + " method on the " + className + " class. Please revert it to the original state.");
+            output = testClass.callMethod(getNumEyesMethodName, classInstance);
+            _assertEquals(eyes, output,
+                    "You have modified the provided " + getNumEyesMethodName + " method on the " + className + " class. Please revert it to the original state.");
+            output = testClass.callMethod(setMyNameMethodName, new Object[][]{{updateName, String.class}}, classInstance);
+            _assertEquals(updateName, testClass.getFieldValue(classInstance, myNameAttributeName),
+                    "You have modified the provided " + setMyNameMethodName + " method on the " + className + " class. Please revert it to the original state.");
+            assertNull(output,
+                    "You have modified the provided " + setMyNameMethodName + " method on the " + className + " class. Please revert it to the original state.");
+            output = testClass.callMethod(setNumEyesMethodName, new Object[][]{{updateEyes, int.class}}, classInstance);
+            _assertEquals(updateEyes, testClass.getFieldValue(classInstance, numEyesAttributeName),
+                    "You have modified the provided " + setNumEyesMethodName + " method on the " + className + " class. Please revert it to the original state.");
+            assertNull(output,
+                    "You have modified the provided " + setNumEyesMethodName + " method on the " + className + " class. Please revert it to the original state.");
+        }
+    }
+
+    private static Stream<Arguments> inputProvider() {
+        return Stream.of(
+                Arguments.of("Sarah", 0, "John", 1),
```

Comment:
You should also have some test cases with larger numbers, and also where the donor has fewer eyes than the recipient.